tests/blocking_ops.rs

```diff
@@ -86,4 +94,88 @@ mod integration_tests {
 
         Ok(())
     }
+
+    // Show that SDK operations timeout correctly using BlockingIronOxide
+    #[test]
+    fn initialize_with_timeout() -> Result<(), IronOxideErr> {
+        let account_id: UserId = create_id_all_classes("").try_into()?;
+        BlockingIronOxide::user_create(
+            &gen_jwt(Some(account_id.id())).0,
+            USER_PASSWORD,
+            &UserCreateOpts::new(false),
+        )?;
+        let device = BlockingIronOxide::generate_new_device(
+            &gen_jwt(Some(account_id.id())).0,
+            USER_PASSWORD,
+            &Default::default(),
+            None,
+        )?
+        .into();
+
+        // set a timeout that is unreasonably small
+        let duration = Duration::from_millis(5);
+        let config = IronOxideConfig {
+            sdk_operation_timeout: Some(duration),
+            ..Default::default()
+        };
+
+        let result = ironoxide::blocking::initialize(&device, &config);
+        let err_result = result.unwrap_err();
+
+        assert_that!(&err_result, is_variant!(IronOxideErr::OperationTimedOut));
+        assert_that!(
+            &err_result,
+            has_structure!(IronOxideErr::OperationTimedOut {
+                operation: eq(SdkOperation::InitializeSdk),
+                duration: eq(*duration)
+            })
+        );
+
+        Ok(())
+    }
+
+    #[test]
+    fn rotate_all_with_timeout() -> Result<(), IronOxideErr> {
+        let account_id: UserId = create_id_all_classes("").try_into()?;
+        BlockingIronOxide::user_create(
+            &gen_jwt(Some(account_id.id())).0,
+            USER_PASSWORD,
+            &UserCreateOpts::new(true),
+        )?;
+        let device = BlockingIronOxide::generate_new_device(
+            &gen_jwt(Some(account_id.id())).0,
+            USER_PASSWORD,
+            &Default::default(),
+            None,
+        )?
+        .into();
+
+        // set a timeout that is unreasonably small
+        let duration = Duration::from_millis(5);
+        let config = IronOxideConfig {
+            sdk_operation_timeout: Some(duration),
+            ..Default::default()
+        };
+
+        if let InitAndRotationCheck::RotationNeeded(bio, to_rotate) =
+            ironoxide::blocking::initialize_check_rotation(&device, &Default::default())?
+        {
+            let result = bio.rotate_all(&to_rotate, USER_PASSWORD, Some(Duration::from_millis(10)));
+
+            let err_result = result.unwrap_err();
+
+            assert_that!(&err_result, is_variant!(IronOxideErr::OperationTimedOut));
+            assert_that!(
+                &err_result,
+                has_structure!(IronOxideErr::OperationTimedOut {
+                    operation: eq(SdkOperation::RotateAll),
+                    duration: eq(*duration)
+                })
+            );
+            Ok(())
+        } else {
+            // error type here is bogus, but will cause the test to fail
+            Err(IronOxideErr::InitializeError)
```

Comment:
This could be replaced with something a bit more telling like `panic!("Rotation should have been required")` and not require a bogus error. Also I don't see the benefit of `if let` here compared to a `match`, but I know that's a discussion that several of us have been having.

Reply:
I think there are usecases for both, and there are some where they are both equally valid. In this case I only want to handle one case which is great for if let. I do like the `panic` suggestion.